### PR TITLE
Hybrid Router test Activity

### PR DESCRIPTION
**Pull request by olegzil**
## Description
This PR is adds an activity that exercises the Hybrid router
Please include a brief summary of the change and which issue is fixed (e.g., fixes [#issue](link))

## Testing
- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ x] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ x] New and existing unit tests pass locally with my changes

## Checklist

- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

**Comment by Lebedinsky** [comment]
I think we need to wait until we merge this PR (https://github.com/mapbox/mapbox-navigation-android/pull/2382) and after we need to update this one, am I right?

**Comment by olegzil** [comment]
> I think we need to wait until we merge this PR (#2382) and after we need to update this one, am I right?

Yes. I'll just make sure this branch is up to date for now.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -4,6 +4,9 @@
     <string name="title_mock_navigation">Mock Navigation</string>
     <string name="description_mock_navigation">Mock a navigation session using a mock location engine.</string>
 
+    <string name="title_component_hybrid_hav">Hybrid Navigation</string>
```
Comment (Guardiola31337) [inline app/src/main/res/values/strings.xml]: NIT These `string` resources are not needed in the `app` module, we should remove.

Comment (olegzil) [reply app/src/main/res/values/strings.xml]: 👍

File: libdirections-onboard/src/test/java/com/mapbox/navigation/route/onboard/MainCoroutineRule.kt [hunk 2]
```diff
@@ -3,31 +3,37 @@ package com.mapbox.navigation.route.onboard
...
-import org.junit.rules.TestWatcher
+import org.junit.rules.TestRule
 import org.junit.runner.Description
+import org.junit.runners.model.Statement
 
 @ExperimentalCoroutinesApi
-class MainCoroutineRule(
-    val testDispatcher: TestCoroutineDispatcher = TestCoroutineDispatcher()
-) : TestWatcher() {
+class MainCoroutineRule : TestRule {
```
Comment (Guardiola31337) [inline libdirections-onboard/src/test/java/com/mapbox/navigation/route/onboard/MainCoroutineRule.kt]: Aren't these changes the same introduced in https://github.com/mapbox/mapbox-navigation-android/pull/2409?

File: examples/src/main/java/com/mapbox/navigation/examples/MainActivity.java [hunk 3]
```diff
@@ -73,6 +74,11 @@ protected void onCreate(Bundle savedInstanceState) {
         getString(R.string.description_offboard_router_kotlin),
         OffboardRouterActivityKt.class
       ),
+      new SampleItem(
+        getString(R.string.title_component_hybrid_navigation),
+        getString(R.string.description_hybrid_router),
+        HybridNavigationActivity.class
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/MainActivity.java]:
```suggestion
        HybridRouterActivityKt.class
```

File: examples/src/main/java/com/mapbox/navigation/examples/MainActivity.java [hunk 4]
```diff
@@ -73,6 +74,11 @@ protected void onCreate(Bundle savedInstanceState) {
         getString(R.string.description_offboard_router_kotlin),
         OffboardRouterActivityKt.class
       ),
+      new SampleItem(
+        getString(R.string.title_component_hybrid_navigation),
+        getString(R.string.description_hybrid_router),
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/MainActivity.java]:
```suggestion
        getString(R.string.description_hybrid_router_kotlin),
```

File: examples/src/main/java/com/mapbox/navigation/examples/MainActivity.java [hunk 5]
```diff
@@ -73,6 +74,11 @@ protected void onCreate(Bundle savedInstanceState) {
         getString(R.string.description_offboard_router_kotlin),
         OffboardRouterActivityKt.class
       ),
+      new SampleItem(
+        getString(R.string.title_component_hybrid_navigation),
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/MainActivity.java]:
```suggestion
        getString(R.string.title_component_hybrid_router_kotlin),
```

File: libdirections-onboard/src/test/java/com/mapbox/navigation/route/onboard/MainCoroutineRule.kt (hunk 2 above)
Comment (abhishek1508) [reply libdirections-onboard/src/test/java/com/mapbox/navigation/route/onboard/MainCoroutineRule.kt]: Let's remove this file. @Guardiola31337 pointed it out right.

**Comment by Guardiola31337** [comment]
Superseded by https://github.com/mapbox/mapbox-navigation-android/pull/2740

Closing this one.

cc @korshaknn

**Pull request closed without merging**